### PR TITLE
feat(jest): support Jest config in different formats

### LINKED ISSUE #1057: @angular-builders/jest - Support jest.config in typescript
**Is your feature request related to a problem? Please describe.**
According https://jestjs.io/docs/configuration it's possible to use a Typescript jest-config if `ts-node` is installed, but I tried it and I got
`An unhandled exception occurred: Cannot use import statement outside a module`

**Describe the solution you'd like**
Support a typescript option for jest config file.

**Describe alternatives you've considered**
I'm trying to customize the `jsdom` options and I just cannot do it with plain javascript it seems:
```
const VirtualConsole = new require('jsdom/lib/jsdom/virtual-console');

const virtualConsole = (new VirtualConsole()).sendTo(console, { omitJSDOMErrors: true });
console.log(virtualConsole)

process.env.TZ = 'UTC';

module.exports = {
  testRunner: 'jest-jasmine2',
  testEnvironment: 'jsdom',
  testEnvironmentOptions: { virtualConsole },
```

### PR DESCRIPTION
**Pull request by arturovt**
Closes #1057

### CODE REVIEW

File: packages/jest/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// The type is used to replace the `any` type to be more explicit with return types.
+// We can't use the `Config` type from `jest` since this might be a breaking change.
```
Comment (just-jeb) [inline packages/jest/src/types.ts:2]: Why would it be a breaking change?

**Review by just-jeb — APPROVED** [review]

File: packages/jest/src/types.ts (hunk 1 above)
Comment (arturovt) [reply packages/jest/src/types.ts:2]: It doesn’t ensure to be a breaking change. I’m just _afraid_ it to be a breaking change.

**Comment by arturovt** [comment]
I've updated the type to point to `jest.Config`.

File: packages/jest/src/types.ts (hunk 1 above)
Comment (just-jeb) [reply packages/jest/src/types.ts]: Yeah I understand, I'm just not sure why 😁. Do you have a specific scenario in mind?

Comment (arturovt) [reply packages/jest/src/types.ts]: I just thought initially that someone may `require` this package and might have issues with typings.

Comment (arturovt) [reply packages/jest/src/types.ts]: But I ain't sure if someone will ever `require` it.

**Pull request merged**